pkg/security/security_profile/secprofs.go

```diff
@@ -330,10 +389,17 @@ func (m *Manager) unlinkProfileMap(profile *profile.Profile, workload *tags.Work
 		return
 	}
 
-	if err := m.securityProfileMap.Delete([]byte(workload.ContainerID)); err != nil {
-		seclog.Errorf("couldn't unlink workload %s (selector: %s) with profile %s: %v", workload.ContainerID, workload.Selector.String(), profile.Metadata.Name, err)
+	if workload.ContainerID != "" {
+		if err := m.securityProfileMap.Delete([]byte(workload.ContainerID)); err != nil {
+			seclog.Errorf("couldn't unlink workload %s (selector: %s) with profile %s: %v", workload.ContainerID, workload.Selector.String(), profile.Metadata.Name, err)
+		}
+		seclog.Infof("workload %s (selector: %s) successfully unlinked from profile %s", workload.ContainerID, workload.Selector.String(), profile.Metadata.Name)
+	} else if workload.CGroupContext.CGroupID != "" {
+		if err := m.securityProfileMap.Delete([]byte(workload.CGroupContext.CGroupID)); err != nil {
+			seclog.Errorf("couldn't unlink cgroup %s (selector: %s) with profile %s: %v", workload.CGroupContext.CGroupID, workload.Selector.String(), profile.Metadata.Name, err)
+		}
+		seclog.Infof("cgroup %s (selector: %s) successfully unlinked from profile %s", workload.CGroupContext.CGroupID, workload.Selector.String(), profile.Metadata.Name)
```

Comment:
There is multiple part where we are manually checking in which case we are. Would we instead for code clarity add some helpers? Here for example we could replace
`m.securityProfileMap.Delete([]byte(workload.ContainerID));
`by
`m.securityProfileMap.Delete([]byte(workload.GetSelectorKey()));` (something like that)
Maybe this way we could simplify other parts of the code, WDYT?